backend/battle/battles/battle.py

```diff
@@ -1,29 +1,42 @@
-# from ..models import Battle
-# from .rounds import roundRunning
-# from pokemon.helpers import get_pokemon_from_api
-
-
-# def battleRunning(poke_id, pokemons):
-#     battle_info = Battle.objects.get(id=poke_id)
-#     poke_info1_creator = get_pokemon_from_api(battle_info.pk1_creator)
-#     poke_info2_creator = get_pokemon_from_api(battle_info.pk2_creator)
-#     poke_info3_creator = get_pokemon_from_api(battle_info.pk3_creator)
-#     # poke_info_creator_list = [poke_info1_creator, poke_info2_creator, poke_info3_creator]
-#     poke_info1_opponent = get_pokemon_from_api(pokemons[0])
-#     poke_info2_opponent = get_pokemon_from_api(pokemons[1])
-#     poke_info3_opponent = get_pokemon_from_api(pokemons[2])
-#     round_one = [poke_info1_creator, poke_info1_opponent]
-#     round_two = [poke_info2_creator, poke_info2_opponent]
-#     round_three = [poke_info3_creator, poke_info3_opponent]
-#     winner = "nobody"
-#     opponent = 0
-#     creator = 0
-#     battle_rounds = [round_one, round_two, round_three]
-#     for battle_round in battle_rounds:
-#         winner = "creator"
-
-#     return winner
-
-# def message_error():
-#     message = "ERROR: PKNs you selected sum more than 600 points, please choose again"
-#     return message
+from battle.battles.rounds import run_round
+from battle.battles.email import result_battle
+
+
+def run_battle(battle):
+    teams = battle.teams.all()
+    assert teams.count() == 2, "Unexpected error. Battle has more than two teams."
```

Comment:
Just a small optimization you may want to learn here.
If you do `battle.teams.all()` then `(teams[0], teams[1])` you do 1 DB query. But `teams.count()` does another DB query.

Since you're going to fetch anyway the team items (0 and 1), you can always ensure 1 DB query only by doing this:

```
teams = list(battle.teams.all())
assert len(teams) == 2, "Unexpected error. Battle has more than two teams."
team_winner = get_winner_for(teams[0], teams[1])
```

Reply:
Great! I didn't know about it. I'm going to improve this :) Thank you